docs/advanced/styling.md

```diff
@@ -8,20 +8,337 @@ has_toc: false
 | tasks-urgency                  | This is applied to the SPAN that wraps the urgency score if displayed on the task.                              |
-| task-list-item-checkbox        | This is applied to the INPUT element for the task.                                                              |
 | tasks-group-heading            | This is applied to H4, H5 and H6 group headings                                                                 |
 
 {: .released }
 `tasks-group-heading` was introduced in Tasks 1.6.0.<br>
 `plugin-tasks-query-explanation` was introduced in Tasks 1.19.0.
+
+## Examples
+
+The following examples can be used as [Obsidian CSS snippets](https://help.obsidian.md/How+to/Add+custom+styles#Use+Themes+and+or+CSS+snippets).
+
+**Tip:** the following examples use CSS variables (`--var(...)`) provided by Obsidian instead of concrete color codes to maximize the chance that the result will be in-line with your chosen theme. You may of course use specific colors if so you choose.
+
+### General Formatting
+
+Making tags, internal links and the recurrence rules of tasks to appear in gray:
+
+```css
+.tasks-list-text a.tag {
+ color: var(--list-marker-color);
+}
+.tasks-backlink a.internal-link {
+ color: var(--list-marker-color);
+}
+.task-recurring {
+ color: var(--list-marker-color);
+}
+```
+
+### Priority as a Checkbox Color
+
+The following rules remove the Tasks priority emoticon and render the tasks' checkboxes in red, blue and orange according to the tasks' priority:
+
+```css
+.task-priority-high input[type=checkbox] {
+ box-shadow: 0px 0px 2px 2px var(--color-red);
+ border-color: var(--color-red);
+}
+.task-priority-low input[type=checkbox] {
+ box-shadow: 0px 0px 2px 2px var(--color-blue);
+ border-color: var(--color-blue);
+}
+.task-priority-medium input[type=checkbox] {
+ box-shadow: 0px 0px 2px 2px var(--color-orange);
+ border-color: var(--color-orange);
+}
+/* This part removes the regular priority emoticon */
+span.task-priority {
+ display: none;
+}
+```
+
+### Colors for Due Today and Overdue
+
+The following rules mark 'today' due dates as blue and past due dates as red:
+
+```css
+/* A special color for the 'due' component if it's for today */
+.task-due.task-due-today span {
+ background: var(--code-property);
+ border-radius: 10px;
+ padding: 2px 8px;
+}
+/* A special color for overdue due dates */
+[class*="past-"] .task-due span {
+ background: var(--color-pink);
+ border-radius: 10px;
+ padding: 2px 8px;
+}
+```
+
+### Circle Checkboxes
+
+The following renders checkboxes as circles instead of squares:
+
+```css
+ul > li.plugin-tasks-list-item .task-list-item-checkbox {
+     margin-inline-start: 0;
+  margin: 5px 2px;
+  border-radius: 50%;
+}
+```
+
+### Grid Layout
+
+The following organizes the task structure into a 3-line grid, on which the description is in the first line and the various components are on the 2nd:
```

Comment:
I wonder whether it might be helpful to select on whether the query is in short mode or not?

Reply:
I think this is actually a good time to discuss this, because it's a case for the usefulness of additional CSS classes that describe the query itself. I'll add classes that will allow different styling according to query options, e.g. not use the grid layout when the query has short mode turned on.